### DISCUSSION TITLE
Adding support to deepseek?

### DISCUSSION THREAD

**Discussion by PucaVaz** [vanna-ai/vanna#709] [Ideas]
Looks like a good LLM, and the implementation don't seem too hard to do, we can use basic the same code of the openai

https://api-docs.deepseek.com/

**Comment by zainhoda** [comment]
Since this already uses the OpenAI client, you can do something like this:

```python
from vanna.openai import OpenAI_Chat
from vanna.chromadb import ChromaDB_VectorStore

from openai import OpenAI

deepseek_client = OpenAI(api_key="<DeepSeek API Key>", base_url="https://api.deepseek.com")

class MyVanna(ChromaDB_VectorStore, OpenAI_Chat):
    def __init__(self, config=None):
        ChromaDB_VectorStore.__init__(self, config=config)
        OpenAI_Chat.__init__(self, client=deepseek_client, config=config) # Your deepseek client is used here

vn = MyVanna(config={'model': 'deepseek-chat'})
```

**Reply by PucaVaz** [reply]
Got it, thanks Zain!

**Comment by jagdishgg** [comment]
For local llm we can use ollama and use ollama api. I am yet to try this
Do you think deepseek coder v2 will be faster and more accurate for generation of sql ? Can I use with ollama and vanna
